python/ray/data/read_api.py

```diff
@@ -338,11 +338,15 @@ def read_datasource(
     block_list.compute_first_block()
     block_list.ensure_metadata_for_first_block()
 
-    return Dataset(
-        ExecutionPlan(block_list, block_list.stats(), run_by_consumer=False),
-        0,
-        False,
+    ds = Dataset(
+        plan=ExecutionPlan(block_list, block_list.stats(), run_by_consumer=False),
+        epoch=0,
+        lazy=True,
     )
+    # NOTE: call ds._plan.execute() here to keep the existing ds.schema() and other
+    # APIs are still working. This is NOT triggering execution for all input blocks.
+    ds._plan.execute()
```

Comment:
This seems odd, can't we have `ds.schema()` trigger (minimal) computation instead?

Reply:
Actually this is not needed, and should be covered by https://github.com/ray-project/ray/pull/31286#discussion_r1063015792 . So removed here.